src/main/java/seedu/duke/logic/BudgetLogic.java

```diff
@@ -29,22 +32,22 @@ public BudgetLogic(Budget budget, AppUi ui) {
      * Sets the budget if it has not been set. If the budget is already set,
      * prompts the user to confirm whether they want to modify it.
      */
-    public void setBudget() {
+    public void setBudget(FinancialList financialList) throws FinanceBuddyException {
         if (!budget.isBudgetSet()) {
             ui.displaySetBudgetMessage();
-            handleSetBudget();
+            handleSetBudget(financialList);
         } else {
             System.out.println("Your current budget is: " + budget.getBudgetAmount());
             System.out.println("Would you like to modify your budget? (yes/no)");
-            handleSetBudget();
+            handleSetBudget(financialList);
         }
     }
 
     /**
      * Handles the process of setting the budget amount by receiving input from the user.
      * Validates that the input is a non-negative number before setting the budget.
      */
-    public void handleSetBudget() {
+    public void handleSetBudget(FinancialList financialList) throws FinanceBuddyException {
         String input = ui.getUserInput();
         if (input.equalsIgnoreCase("yes")) {
```

Comment:
Perhaps would be good to add feedback if invalid entry/user enters no, but I understand that is Verity's part?